File: app/src/main/java/com/tencent/component/textspan/SpanFragment.java
package com.tencent.component.textspan;

import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.text.SpannableStringBuilder;
import android.text.Spanned;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.tencent.component.R;
import com.tencent.component.common.DensityUtils;

public class SpanFragment extends Fragment {

    private TextView mTextView;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_span, container, false);
        mTextView = view.findViewById(R.id.textview);
        return view;
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
//        ColorStateList colorStateList = new ColorStateList(new int[][]{{android.R.attr.state_selected}, {0}},
//                new int[]{Color.RED, Color.BLACK});
//        mTextView.setTextColor(colorStateList);
//        mTextView.setSelected(true);

        CharSequence text = mTextView.getText();

        SpannableStringBuilder builder = new SpannableStringBuilder();
        builder.append(text);
        int start = text.length();
        builder.append(" ");
        Drawable drawable = getResources().getDrawable(R.mipmap.ic_launcher_round);
        MyImageSpan span = new MyImageSpan(drawable);
        drawable.setBounds(0, 0, DensityUtils.dip2px(getContext(), 16),
                DensityUtils.dip2px(getContext(), 12                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                ));
        builder.setSpan(span, start, start+1, Spanned.SPAN_EXCLUSIVE_EXCLUSIVE);
        mTextView.setText(builder);

    }
}

File: app/src/main/java/com/tencent/component/databinding/SpanFragment.kt
package com.tencent.component.databinding

import android.os.Bundle
import android.support.v4.app.Fragment
import android.text.Spannable
import android.text.SpannableStringBuilder
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.tencent.component.R

class SpanFragment: Fragment() {

    @Override
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_span, container, false)
        val textView = view.findViewById<TextView>(R.id.textview)
        val text = "我怕的打好看的撒环境的按时的及哦啊的我怕的打好看的撒环境的按时的及哦啊的我怕的打好看的撒环境的按时的及哦啊的"
//        val span = UrlImageSpan(this.context, "http://shp.qpic.cn/wg_hpic/0/avatar_1511611390494/0", textView)
//        val builder = SpannableStringBuilder(text)
//        builder.setSpan(span, 0, 1, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        textView.text = text
        return view
    }
}